cmd/controller/controller.go

```diff
@@ -190,38 +283,40 @@ func (c *Controller) updateRelease(key string) error {
 
 	helmObj := obj.(*helmCrdV1.HelmRelease)
 
-	// FIXME: make configurable
-	keyring := "/keyring/pubring.gpg"
-
-	dl := downloader.ChartDownloader{
-		HelmHome: settings.Home,
-		Out:      os.Stdout,
-		Keyring:  keyring,
-		Getters:  getter.All(settings),
-		Verify:   downloader.VerifyNever, // FIXME
-	}
-
 	repoURL := helmObj.Spec.RepoURL
 	if repoURL == "" {
 		// FIXME: Make configurable
 		repoURL = defaultRepoURL
 	}
 
-	certFile := ""
-	keyFile := ""
-	caFile := ""
-	chartURL, err := repo.FindChartInRepoURL(repoURL, helmObj.Spec.ChartName, helmObj.Spec.Version, certFile, keyFile, caFile, getter.All(settings))
+	authHeader := ""
+	if helmObj.Spec.Auth.Header != nil {
+		namespace := os.Getenv("POD_NAMESPACE")
+		if namespace == "" {
+			namespace = defaultNamespace
+		}
+		secret, err := c.kubeClient.Core().Secrets(namespace).Get(helmObj.Spec.Auth.Header.SecretKeyRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		var b bytes.Buffer
+		b.Write(secret.Data[helmObj.Spec.Auth.Header.SecretKeyRef.Key])
+		authHeader = b.String()
```

Comment:
I think you're just trying to convert the secret data (`[]bytes`) into a `string` here.  This can be done directly with:
```go
data := secret.Data[helmObj.Spec.Auth.Header.SecretKeyRef.Key]
authHeader = string(data)
```